l10n_br_account_product/models/account.py

```diff
@@ -0,0 +1,278 @@
+        result_ipi = self._compute_tax(cr, uid, specific_ipi, result['total'],
+                                       product, quantity, precision, base_tax)
+        totaldc += result_ipi['tax_discount']
+        calculed_taxes += result_ipi['taxes']
+        for ipi in result_ipi['taxes']:
+            ipi_value += ipi['amount']
+
+        # Calcula ICMS
+        specific_icms = [tx for tx in result['taxes']
+                         if tx['domain'] == 'icms']
+        if fiscal_position and fiscal_position.asset_operation:
+            total_base = result['total'] + insurance_value + \
+                freight_value + other_costs_value + ipi_value
+        else:
+            total_base = result['total'] + insurance_value + \
+                freight_value + other_costs_value
+
+        result_icms = self._compute_tax(
+            cr,
+            uid,
+            specific_icms,
+            total_base,
+            product,
+            quantity,
+            precision,
+            base_tax)
+        totaldc += result_icms['tax_discount']
+        calculed_taxes += result_icms['taxes']
+        if result_icms['taxes']:
+            icms_value = result_icms['taxes'][0]['amount']
+
+        # Calcula ICMS ST
+        specific_icmsst = [tx for tx in result['taxes']
+                           if tx['domain'] == 'icmsst']
+        result_icmsst = self._compute_tax(cr, uid, specific_icmsst,
+                                          result['total'], product,
+                                          quantity, precision, base_tax)
+        totaldc += result_icmsst['tax_discount']
+        if result_icmsst['taxes']:
+            icms_st_percent = result_icmsst['taxes'][0]['percent']
+            icms_st_percent_reduction = result_icmsst[
+                'taxes'][0]['base_reduction']
+            icms_st_base = round(((result['total'] + ipi_value) *
+                                 (1 - icms_st_percent_reduction)) *
+                                 (1 + result_icmsst['taxes'][0]['amount_mva']),
+                                 precision)
+            icms_st_base_other = round(
+                ((result['total'] + ipi_value) * (
+                    1 + result_icmsst['taxes'][0]['amount_mva'])),
+                precision) - icms_st_base
+            result_icmsst['taxes'][0]['total_base'] = icms_st_base
+            result_icmsst['taxes'][0]['amount'] = round(
+                (icms_st_base * icms_st_percent) - icms_value, precision)
+            result_icmsst['taxes'][0]['icms_st_percent'] = icms_st_percent
+            result_icmsst['taxes'][0][
+                'icms_st_percent_reduction'] = icms_st_percent_reduction
+            result_icmsst['taxes'][0][
+                'icms_st_base_other'] = icms_st_base_other
+
+            if result_icmsst['taxes'][0]['percent']:
+                calculed_taxes += result_icmsst['taxes']
+
+        # Estimate Taxes
+        if fiscal_position and fiscal_position.asset_operation:
```

Comment:
We should change this to the "ind_final" flag that came with nfe 3.10:
https://github.com/odoo-brazil/l10n-brazil/issues/120